gobblin-config-management/src/main/java/gobblin/config/utils/PathUtils.java

```diff
@@ -0,0 +1,87 @@
+package gobblin.config.utils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.hadoop.fs.Path;
+
+
+/*
+ * Used to resolve the hadoop 1 -> 2 issue
+ */
+public class PathUtils {
+
+  public static Path getParent(URI uri) {
+    if (uri == null)
+      return null;
+
+    String pStr = uri.getPath();
+    if (pStr.length() == 0)
+      return null;
+
+    Path p = (new Path(uri.getPath())).getParent();
+    return p;
+  }
+
+  public static URI getParentURI(URI uri) {
+    String pStr = uri.getPath();
+    if (pStr.length() == 0)
+      return null;
+
+    Path p = (new Path(uri.getPath())).getParent();
+    try {
+      return new URI(uri.getScheme(), uri.getAuthority(), p.toString(), uri.getQuery(), uri.getFragment());
+    } catch (URISyntaxException e) {
+      // Should not come here
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  /**
+   * @param parent - parent URI
+   * @param child - descendant URI
+   * @return true if o2 is the descendant of o1 
+   */
+  public static boolean checkDescendant(URI parent, URI child) {
```

Comment:
We have at least one other implementation of this method in gobblin-data-management (it is also simpler). We should just use one in gobblin-util.